certstore/certstore.go

```diff
@@ -0,0 +1,120 @@
+package certstore
+
+import (
+	"time"
+
+	"crypto/x509"
+	"database/sql"
+
+	cferr "github.com/cloudflare/cfssl/errors"
+	"github.com/cloudflare/cfssl/log"
+	_ "github.com/mattn/go-sqlite3" // Initialize SQLite without bringing it in scope
+)
+
+// Certificate represents a certificate in the certstore
+type Certificate struct {
+	AsPEM            string
+	Serial           string
+	SignedAt         time.Time
+	Expiration       time.Time
+	RevokedAt        *time.Time
+	RevocationReason sql.NullInt64
+}
+
+var certificatesSchema = `
```

Comment:
my take on the certificate structure
https://github.com/cloudflare/cfssl/commit/8810ff8aba1bb426b8dca24fa7f54e0927cc65b6